lib/constraint_solver/generalization.ml

```diff
@@ -405,11 +706,54 @@ let generalize_young_region ~state (young_region : Young_region.t) =
          assert (Tree.Level.(r.level = young_level));
          (* Make the type generic *)
          Type.generalize type_;
+         (* Cannot generalize unresolved svar *)
+         (match Type.status type_, Type.inner type_ with
+          | Generic, Var (Empty_one_or_more_handlers _) -> raise Cannot_unsuspend_generic
+          | _ -> ());
          true)))
   in
   [%log.global.debug "Generics for young region" (generics : Type.t list)];
-  young_region.region.types <- [];
-  [%log.global.debug "Updated region" (young_region.region : Type.region)]
+  (* Deal with partial generics that can be generalized to full generics *)
+  [%log.global.debug
+    "Partially generalizing remaining partial generics" (generics : Type.t list)];
+  List.iter generics ~f:(fun generic ->
+    [%log.global.debug "Partially generalizing generic" (generic : Type.t)];
+    Type.partial_generalize generic ~f:(fun (guard, instance) ->
+      [%log.global.debug
+        "Generalizing the partial generic, visiting instance"
+          (generic : Type.t)
+          (guard : Guard.t)
+          (instance : Type.t)];
+      (* The partial generic that links to [instance] has been fully generalized :) *)
+      let curr_region = Type.region_exn ~here:[%here] instance in
+      (* [curr_region] is a sibling or parent of the current region. It cannot be a decsendant *)
+      visit_region ~state curr_region;
+      (* Perform a partial copy on the generic to ensure the instance has the generalized
+         structure and then unify *)
+      let copy = partial_copy ~state ~curr_region generic in
+      (* NOTE: Scheduler jobs that are queued by [unify] and [remove_guard] only visit siblings or parents. *)
+      unify ~state ~curr_region copy instance;
+      (* Remove the guard *)
+      remove_guard ~state instance guard);
+    [%log.global.debug "Partially generalized generic" (generic : Type.t)]);
```

Comment:
Inefficiency: This exhibits quadratic behaviour if we generalize from the bottom up for a given partially generic type. 
This is because we will copy (fully) generic types deeply in `partial_copy` and if we generalize from the bottom up, we copy a quadratic number of nodes for a given type. 

We can address this (in a follow up PR) by updating the status of the partial generics *after all* copying is performed.